Spec: Update mediaType to type for catalogEntry and add known types (#37)
Agent-Card/ai-card: adr/0014-media-type-to-type.md

Accepted

## Date
2026-04-30 2026-05-14

**Participants:** Pamela Dingle (Microsoft), Sam Betts (Cisco), Junjie Bu (Google), Darrel Miller (Microsoft), Alan Blount (Google), Srinivas Krishnan (Google), Krishna Thota (Google)

## Context
The `ai-catalog` specification originally used `mediaType` to identify the type of AI artifact in a catalog entry, intending to leverage standard IANA media types. However, in practice, we are using a closed, fixed set of types specific to the AI Catalog ecosystem (e.g., A2A agent cards, MCP servers, nested catalogs). Using the name `mediaType` implies support for any valid IANA media type and dynamic content negotiation, which is not the intent and can be confusing for implementers.

Furthermore, we need to support combined types that specify both the artifact type and the payload format (e.g., `application/ai-skills+tgz`).

## Decision
We will rename the `mediaType` field to `type` in the `CatalogEntry` schema.
The `type` field is an open text format, so any string value is accepted. The following are recognized "known types" in the ecosystem:
- `application/a2a-agent-card+json`
- `application/mcp-server+json`
- `application/ai-catalog+json`

> - `application/ai-skills+tgz`

Reviewer: There is a +gzip but there is not +tgz that is registered and so currently it would not be possible to register this media type.

Author: Updated the adr and spec. Added the +gzip but removed +tgz.

- `application/ai-skills+zip`

> - `application/ai-skills+md`

Reviewer: As there is no registered suffix, it would not be possible at present to register this media type.

Author: Changed this `application/ai-skills+md` to `text/markdown; profile=ai-skill`

Additionally, the `Attestation` object in the `trustManifest` schema had both a `type` field (identifying the attestation type, e.g., `SOC2-Type2`) and a `mediaType` field (indicating the format of the attestation document, e.g., `application/pdf`). This created terminology inconsistency and confusion for implementers who had to negotiate two separate typing fields within the same attestation metadata.

## Decision
We will make the following schema simplifications:

### 1. Rename `mediaType` to `type` in `CatalogEntry`
We will rename the `mediaType` field to `type` in the `CatalogEntry` schema.
The `type` field is an open text format, so any string value is accepted. To ensure clarity of ownership and governance, the recognized "known types" are partitioned into core protocol types and integrated third-party/ecosystem types:

#### Core Protocol Types (Governed by the AI Catalog WG)
- `application/ai-catalog+json` — nested AI Catalog
- `application/agent-card+json` — reserved for a generic Agent Card format

#### Integrated Ecosystem & Third-Party Types (Governed externally)
- `application/a2a-agent-card+json` — A2A Agent Card

> - `application/mcp-server+json` — MCP Server Card

Reviewer 2: ```suggestion
- `application/mcp-server-card+json` — MCP Server Card
```

- `application/ai-skills+zip` — AI Skill bundle in a ZIP archive
- `application/ai-skills+gzip` — AI Skill bundle in a gzipped tarball

> - `text/markdown; profile=ai-skill` — AI Skill defined in a standard Markdown file

Reviewer 3: Should these be "Agent Skill" (per https://agentskills.io/) instead of "AI Skill"?

```suggestion
- `application/agent-skills+zip` — Agent Skill bundle in a ZIP archive
- `application/agent-skills+gzip` — Agent Skill bundle in a gzipped tarball
- `text/markdown; profile=agent-skill` — Agent Skill defined in a standard Markdown file
```

For any new or custom types not listed here, it is up to the specific client implementation to handle them correctly.

### 2. Remove `mediaType` from Attestations
We will completely remove the `mediaType` field from the `Attestation` object schema under `trustManifest`. Instead of requiring an explicit `mediaType` inside the attestation entry, consumers can resolve or infer the format directly from the HTTP response headers when fetching the attestation `uri`, or from the file extension in the URL path.

## Rationale

### Suffixes and Parameters in `CatalogEntry`
To ensure that our recommended types are fully registerable as official IANA media types, we adhere strictly to the IANA Structured Syntax Suffix registry rules (RFC 6838/6839):
- **Custom Ecosystem Formats**: Types like A2A Agent Cards (`application/a2a-agent-card+json`), MCP Servers (`application/mcp-server+json`), and Nested Catalogs (`application/ai-catalog+json`) use custom domain prefixes combined with the standard, registered `+json` suffix.

> - **AI Skill Bundles**: Packages are represented using custom prefixes combined with standard, registered syntax suffixes for archives/compression: `+zip` (registered under RFC 9559) and `+gzip` (registered under RFC 6839). Unregistered suffixes like `+tgz` are avoided to prevent registration failure.

Reviewer 3: ```suggestion
- **Agent Skill Bundles**: Packages are represented using custom prefixes combined with standard, registered syntax suffixes for archives/compression: `+zip` (registered under RFC 9559) and `+gzip` (registered under RFC 6839). Unregistered suffixes like `+tgz` are avoided to prevent registration failure.
```

> - **Generic Formats (Markdown)**: Standard Markdown is represented by the generic, globally registered `text/markdown` media type. Because using `text/markdown` in isolation does not indicate the document is an AI Skill (unlike custom prefixes), we use the standard IANA `profile` parameter (`profile=ai-skill`) to cleanly declare the semantic profile of the document while remaining 100% standard-compliant.

Reviewer 3: ```suggestion
- **Generic Formats (Markdown)**: Standard Markdown is represented by the generic, globally registered `text/markdown` media type. Because using `text/markdown` in isolation does not indicate the document is an Agent Skill (unlike custom prefixes), we use the standard IANA `profile` parameter (`profile=agent-skill`) to cleanly declare the semantic profile of the document while remaining 100% standard-compliant.
```
